feat(coc): adds the code of conduct

anguspiv opened the pull request:
Adds the code of conduct for group participation

re #1

anguspiv committed:
feat(coc): adds the code of conduct

Adds the code of conduct for group participation

resolves #1

anguspiv commented:
I copied the first section of principles from JS.LA, regarding events and online I was thinking that might be early to add in.

powernerd reviewed (approved):
This looks great. How would you feel about adding something about spam and self-promotion in our communication channels?

anguspiv commented:
@powernerd I do like that idea, I think we can take what JS.LA has and probably rework it a bit. Since it's not only slack, but there will probably other channels (github, etc) that it will need to cover.

I'll add something that might cover what we need

anguspiv committed:
feat(coc): adds section on communication principles

Adds a section to the Code of Conduct to address
communications in regards to recruiting and self
promotion

closes #1
